packages/ui-components/src/infinite-scroll.ts

```diff
@@ -41,6 +41,7 @@ interface InfiniteScrollProps<T> extends Omit<APIResultProps<T>, "params"> {
   resultsComponent?: React.ComponentType<{ data: T[] }>;
   perPage?: number;
   startPage?: number;
+  initialData?: T[];
```

Comment:
Might be useful to describe in a comment why you might want to provide initial data (e.g., server loaded first page). The goal of this might not be obvious unless you had looked at our code in `web`